doc/source/data/dataset-internals.rst

```diff
@@ -64,19 +64,17 @@ Execution
 
 This section covers Dataset execution modes and performance considerations.
 
-Lazy Execution Mode
-~~~~~~~~~~~~~~~~~~~
-
-By default, most Datasets operations are eager, which provides a simpler iterative
-development experience. Datasets also has a lazy execution mode that can offer
-improved performance due to stage fusion optimizations.
-
-Lazy execution mode can be enabled by calling
-:meth:`ds = ds.lazy() <ray.data.Dataset.lazy()>`, which
-returns a Dataset whose all subsequent operations will be lazy. These operations
-won't be executed until the dataset is consumed or
-:meth:`ds.fully_executed() <ray.data.Dataset.fully_executed>` is called to manually
-trigger execution.
+Lazy Execution
+~~~~~~~~~~~~~~
+
+By default, most Datasets operations are lazy, with execution triggered via "sink"
```

Comment:
I think it'll provide a clearer picture for users with an API categorization structure: transformation (all lazy), consumption/sink, fully_executed. This is the mental model users should have when thinking about Data APIs.

And then we talk about exceptions like `union`, `split` are something that exist just today; they will be moved to follow the above categorization.